AWSPinpoint/AWSPinpointNotificationManager.m

```diff
@@ -260,6 +269,54 @@ - (void)addApplicationStateAttributeToEvent:(AWSPinpointEvent *) event
     }
 }
 
+- (AWSPinpointEvent*)buildEventFromUserInfo:(NSDictionary *) userInfo
+                         withPushActionType:(AWSPinpointPushActionType) pushActionType {
+    NSString *eventTypePrefix = [self getEventTypePrefixFromUserInfo:userInfo];
+    NSString *eventTypeSuffix = [self getEventTypeSuffixFromPushActionType:pushActionType];
+    if (!eventTypePrefix || !eventTypeSuffix) {
+        return nil;
+    }
+    NSString *eventType = [NSString stringWithFormat: @"_%@.%@", eventTypePrefix, eventTypeSuffix];
+
+    AWSPinpointEvent *pushNotificationEvent  = [self.context.analyticsClient createEventWithEventType:eventType];
+
+    [self addApplicationStateAttributeToEvent:pushNotificationEvent
+                         withApplicationState:[[UIApplication sharedApplication] applicationState]];
```

Comment:
Coupling `addApplicationStateAttributeToEvent:withApplicationState:` to `buildEventFromUserInfo: withPushActionType:` is confusing to me because it doesn't rely on any of the variables we are passing into the function: `userInfo` and `pushActionType`.  I think it makes sense to move this function back to the respective functions of `recordMessageReceivedEventForNotification:withPushActionType:` and `recordMessageOpenedEventForNotification:withIdentifier:`

Reply:
Will move.